lib/ceedling/defaults.rb

```diff
@@ -237,6 +237,17 @@
     ].freeze
   }
 
+DEFAULT_BACKTRACE_TOOL = {
+  :executable => FilePathUtils.os_executable_ext('gdb').freeze,
+  :name => 'default_backtrace_settings'.freeze,
+  :stderr_redirect => StdErrRedirect::AUTO.freeze,
+  :background_exec => BackgroundExec::NONE.freeze,
+  :optional => false.freeze,
+  :arguments => [
+    '-q --eval-command run --eval-command backtrace --batch --args'
```

Comment:
This is different as in other default tools

Reply:
Hi @Letme ,

I understand your point of view. The main idea from my side, it was to write all args at once because later when end user will want to overwrite the config, we drop default change.

```ruby
gdb_extra_args = @configurator.project_config_hash[:tools_backtrace_settings][:arguments].drop(1)
```

We need to do this, as default logic in implementation is to merge tools requirements, not overwrite them from what I see in the code. 

As the idea was to give other people possibility to overwrite gdb by any other lets say arm-none-abi-gdb, or any other debugger,  we need to drop actual arguments, and defined by end user in project.yml file.

Reply:
Hm. We DO definitely do some merging, but my memory thinks that arguments were being replaced, not merged, because of this reason. I'll do a couple experiments to verify this is true... If it's not, we have this same problem elsewhere... replacing a compiler from gcc to something else should also change all the arguments, for example.

Reply:
I will try dig around it, trying to find some answers why this is happening, but my conclusions base at below test:

project.yml:

```yaml
   :arguments: ["-q", "--eval-command run", "--eval-command backtrace", "--batch", "zumzum"]
```

the default is set to:

```yaml
   :arguments: ["-q", "--eval-command run", "--eval-command backtrace", "--batch", "--args"]
```

The output argument which I receive at the end is:

```yaml
   :arguments: ["-q", "--eval-command run", "--eval-command backtrace", "--batch", "--args", "zumzum"]
```
   
which is not exactly what we would like to see when we use any other debugger. Looks like the arguments are combination without repetition between default and user defined arguments array

Reply:
Hi @mvandervoord , @Letme ,

The mentioned by you problem was fixed. Right now the arguments are overwritten properly. I will be glad if you will find one more time to look at my changes.